Sorting/QuickSort.cpp

```diff
@@ -1,11 +1,15 @@
 /*
     Quick sort
     ----------
-    An efficient, comparison-based, in-place, divide and conquer sorting algorithm.
+    An efficient, comparison-based, in-place, divide and conquer sorting algorithm. It picks
+    an element as pivot, partitions the given array around the picked pivot, recursively
+    calls itself on those partitions and then merges them after sorting.
```

Comment:
Quick sort doesn't "merge" the partitions.

The description should therefore be like: `It randomly picks an element as pivot, partitions the given array around the selected pivot while placing the pivot in its sorted position, and recursively calls itself on those partitions.`